### PR TITLE
feat: can now specify multiple providers in a single call

### LINKED ISSUE #16: Allow using a group of providers at once
#### Perceived Problem

```ts
const ctx = kont()
  .useBeforeAll(a())
  .useBeforeEach(b())
  .useBeforeEach(c())
  .done()
```

What if consumer/producer wants to be able to have a,b,c providers packages up
into one group of providers?

#### Ideas / Proposed Solution(s)

```ts
const ctx = kont()
  .useBeforeAll(a(), b(), c())
  .done()
```

```ts
const group = [a(), b(), c()]

const ctx = kont()
  .useBeforeAll(...group)
  .done()
```

```ts
const ctx = kont()
  .useBeforeAll([a(), b(), c()])
  .done()
```

```ts
const group = [a(), b(), c()]

const ctx = kont()
  .useBeforeAll(group)
  .done()
```

### PR DESCRIPTION
**Pull request by just-be-dev**
closes #16

I wrote tests but they're arguably useless. Should probably improve that to actually _do_ something. Also need to add docs. Maybe should add an API section to the readme?

### CODE REVIEW

File: tests/providers/useBeforeAll.spec.ts [hunk 1]
```diff
@@ -31,6 +31,14 @@ describe('upstream providers can satisfy requirements of downstream providers',
   it('test', noop)
 })
 
+describe('can use multiple providers in a single call', () => {
+  const p1 = provider<{}, NoContext>().before(noop).done()
+  const p2 = provider<{}, NoContext>().before(noop).done()
+  kont().useBeforeAll(p1, p2).done()
+
+  it('test', noop)
```
Comment (jasonkuhrt) [inline tests/providers/useBeforeAll.spec.ts:39]: Check that combined context is available in `ctx`.

File: tests/providers/useBeforeAll.spec.ts [hunk 2]
```diff
@@ -31,6 +31,14 @@ describe('upstream providers can satisfy requirements of downstream providers',
   it('test', noop)
 })
 
+describe('can use multiple providers in a single call', () => {
+  const p1 = provider<{}, NoContext>().before(noop).done()
+  const p2 = provider<{}, NoContext>().before(noop).done()
+  kont().useBeforeAll(p1, p2).done()
```
Comment (jasonkuhrt) [inline tests/providers/useBeforeAll.spec.ts:37]: Have `p2` depend on context from `p1` showing that the context typing works not just across the chain but across a group

**Review by jasonkuhrt — CHANGES_REQUESTED** [review]
Thanks @zephraph 

Fix the lint errors please

> Maybe should add an API section to the readme?

I want this to be served by JSDoc + Paka. Therefore focus your efforts in JSDoc. Paka takes care of putting into a nice website.

File: tests/providers/useBeforeAll.spec.ts (hunk 2 above)
Comment (zephraph) [reply tests/providers/useBeforeAll.spec.ts:37]: Yeah, that didn't work. The typing will be a bit trickier I think.

Comment (jasonkuhrt) [reply tests/providers/useBeforeAll.spec.ts:37]: Yep :)

**Pull request closed without merging**